### PR TITLE
User can ask question

### PR DESCRIPTION
**Pull request by jordantomax**
- Add Standard JS linter to lint on `push` and for `pull request`
- Correct `bootstrap` and `react-bootstrap` version numbers to the latest stable versions
- Update schema
- Update Question page to use new vote interface instead of credible counts directly on the model. This was previously throwing an error
- Add `AskQuestion` component to allow user to ask a new question.
- Abstract global form logic into a React hook called `useForm`. This hook is extremely useful and portable. It brings ease of development and consistency to our forms. It standardizes error handling (global and individual input), loading states, onChange and submit logic. It also adds a custom validator that makes it extremely easy to add validation logic per field type, which I have now done for urls. This logic can now be replaced globally in a single location. You can find these validators in `utils/validation.js`. Please use this hook for all forms going forward.
- Please note that the `name` prop is necessary for the hook to correctly update data
- Replace all existing form logic with new `useForm` hook.
- Remove unnecessary form styling, react-bootstrap should by default handle most styling without the need to extend or wrap elements with styled components.
- Abstract shared component logic of text inputs to `TextInput`
- Abstract shared component logic of forms into a `FullPageForm`

### CODE REVIEW

File: src/styles/Button.js
Comment (jtan381) [inline src/styles/Button.js:8]: Container not sure if should be here. I would rather create it induvial.

**Review by fabianchua6 — DISMISSED** [review]

File: src/views/AskQuestion.js
Comment (jtan381) [inline src/views/AskQuestion.js:20]: no validation weather if the user are login in?

File: src/utils/validation.js
Comment (jtan381) [inline src/utils/validation.js:12]: can not just return true in the end try section? why need check if url? Then what about if url is null?

File: src/components/SubmitAnswerForm.js
Comment (jtan381) [inline src/components/SubmitAnswerForm.js]: Can standardize the label and value. Or else you see "Uncertain" when submitting answer, and "Neither" on the answer card.

**Review by jtan381 — DISMISSED** [review]

File: src/styles/Button.js
Comment (jordantomax) [reply src/styles/Button.js:8]: I think it's good to have it abstracted since it's being used for all form, but maybe could move to another file more specific to forms? Let me know what you think, or if I'm misunderstanding.

File: src/views/AskQuestion.js
Comment (jordantomax) [reply src/views/AskQuestion.js:20]: I'm letting the backend handle authentication validation. Since all error handling is standardized now, you'll see a message (or you should at least!) that you don't have permissions when trying to take an action that requires authentication. Maybe in the future, we can build out some frontend checks, but I feel this is quite good for a start.

File: src/utils/validation.js
Comment (jordantomax) [reply src/utils/validation.js:12]: Yeah... Linter gets angry when you don't assign return value of `new` instantiation to a variable, and then gets angry again if you don't reference a declared variable. Hence the check, it's kind of silly but functionally it should work fine. I don't think the `null` example will ever be a case. If it can't create a url, it will throw an error and fail validation.

I'm not sure if there's a good way to use the `URL` class without new. And other validation methods I checked relied on regex, which is fine, but I thought using `URL` is a bit cleaner to read.

File: src/components/SubmitAnswerForm.js
Comment (jordantomax) [reply src/components/SubmitAnswerForm.js]: Good point, updated.

**Comment by jordantomax** [comment]
@jtan381 Thanks, I've updated the `Uncertain` to `Neither` for consistency and responded to your other comments. Let me know what you think about those. Once all looks good, you can review and approve again since changes to the PR dismiss the stale review.

**Review by jtan381 — APPROVED** [review]

**Pull request merged**
